tools/clang/test/HLSL/workgraph/invalid_node_record_type.hlsl

```diff
@@ -0,0 +1,160 @@
+// RUN: %clang_cc1 -HV 2021 -verify %s
+
+struct RECORD
+{
+  uint a;
+  bool b;
+  float4 c;
+  matrix<float,3,3> d;
+};
+
+struct BAD_RECORD
+{
+  uint a;
+  RECORD rec;
+  SamplerState s; // expected-note+ {{'SamplerState' field declared here}}
+};
+
+struct BAD_RECORD2
+{
+  BAD_RECORD bad;
+};
+
+typedef matrix<float,2,2> f2x2;
+
+typedef SamplerState MySamplerState;
+
+typedef BAD_RECORD2 MyBadRecord;
+
+//==============================================================================
+// Check diagnostics for the various node input/output types
+
+[Shader("node")]
+[NodeLaunch("broadcasting")]
+[NumThreads(8,1,1)]
+[NodeMaxDispatchGrid(8,1,1)]
+void node01(DispatchNodeInputRecord<int> input) // expected-error {{'int' is not valid as a node record type}}
```

Comment:
The message doesn't tell me why `int` isn't valid as a node record type. The old message might not have been clear, but it at least explained that a class or struct type was required. Can we adjust the wording?

Reply:
Can do.